src/hb/ot_layout_gsubgpos.rs

```diff
@@ -240,7 +240,20 @@ impl<'a, 'b> skipping_iterator_t<'a, 'b> {
 
     pub fn next(&mut self, unsafe_to: Option<&mut usize>) -> bool {
         assert!(self.num_items > 0);
-        while self.buf_idx + usize::from(self.num_items) < self.buf_len {
+        // The alternate condition below is faster at string boundaries,
+        // but produces subpar "unsafe-to-concat" values.
+        let mut stop: i32 = self.buf_len as i32 - self.num_items as i32;
+
+        if self
+            .ctx
+            .buffer
+            .flags
+            .contains(BufferFlags::PRODUCE_UNSAFE_TO_CONCAT)
+        {
+            stop = self.buf_len as i32 - 1;
+        }
+
+        while (self.buf_idx as i32) < stop {
```

Comment:
Those are the weird casts I had to do, but since harfbuzz does the same it should be okay?